### PR TITLE
build(npm): migrate to npm

### PR DESCRIPTION
**Pull request by bryanjtc**
Migrates from yarn to npm

### CODE REVIEW

**Comment by skovy** [comment]
I haven't been following the yarn releases, is there a significant benefit to v3? When initially working on this project yarn had many features and nice things npm lacked. However, npm has mostly caught up with the things this project uses. Thoughts on switching to latest npm instead of yarn completely?

**Comment by bryanjtc** [comment]
The only thing that npm still lacks significantly is speed. It still is slower than yarn. Yarn introduced plug n' play and zero installs with v2. Plug n' play get rids of node_modules and zero installs just commits the cache folder. This pr doesn't implement plug n' play or zero installs. I can add it in another pr. V3 improves performance and compability. Ideally when you run yarn install with plug n' play and zero installs, the install time is 0 or a few seconds if there is a package that needs to be build like esbuild and sass.

**Comment by bryanjtc** [comment]
@skovy What are your thoughts? I can still redo the pr to move to npm v9 or pnpm

**Comment by skovy** [comment]
my preference would be npm because it's still the "standard" in the community and this project is fairly standard (or would like to keep it so as much as possible)

**Comment by bryanjtc** [comment]
@skovy done

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,12 @@
...
\ No newline at end of file
+.DS_Store
+
+.pnp.*
+.yarn/*
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
```
Comment (skovy) [inline .gitignore:12]: do we need these? pnp/yarn shouldn't exist in this repo so shouldn't cause any of these files to exist?

**Comment by skovy** [comment]
Apologies for the merge conflicts, Travis has been flakey so I setup GitHub actions, which should be more reliable. That also means the `alpha` branch should work and we can merge into it.

**Comment by bryanjtc** [comment]
I changed the base branch to alpha

**Pull request merged**